Mathlib/Algebra/MvPolynomial/Basic.lean

```diff
@@ -372,31 +374,52 @@ theorem induction_on' {P : MvPolynomial σ R → Prop} (p : MvPolynomial σ R)
     show P (monomial 0 0) from h1 0 0)
     fun _ _ _ _ha _hb hPf => h2 _ _ (h1 _ _) hPf
 
-/-- Similar to `MvPolynomial.induction_on` but only a weak form of `h_add` is required. -/
-theorem induction_on''' {M : MvPolynomial σ R → Prop} (p : MvPolynomial σ R) (h_C : ∀ a, M (C a))
-    (h_add_weak :
-      ∀ (a : σ →₀ ℕ) (b : R) (f : (σ →₀ ℕ) →₀ R),
-        a ∉ f.support → b ≠ 0 → M f → M ((show (σ →₀ ℕ) →₀ R from monomial a b) + f)) :
-    M p :=
-    -- Porting note: I had to add the `show ... from ...` above, a type ascription was insufficient.
-  Finsupp.induction p (C_0.rec <| h_C 0) h_add_weak
-
-/-- Similar to `MvPolynomial.induction_on` but only a yet weaker form of `h_add` is required. -/
-theorem induction_on'' {M : MvPolynomial σ R → Prop} (p : MvPolynomial σ R) (h_C : ∀ a, M (C a))
-    (h_add_weak :
-      ∀ (a : σ →₀ ℕ) (b : R) (f : (σ →₀ ℕ) →₀ R),
-        a ∉ f.support → b ≠ 0 → M f → M (monomial a b) →
-          M ((show (σ →₀ ℕ) →₀ R from monomial a b) + f))
-    (h_X : ∀ (p : MvPolynomial σ R) (n : σ), M p → M (p * MvPolynomial.X n)) : M p :=
-    -- Porting note: I had to add the `show ... from ...` above, a type ascription was insufficient.
-  induction_on''' p h_C fun a b f ha hb hf =>
-    h_add_weak a b f ha hb hf <| induction_on_monomial h_C h_X a b
-
-/-- Analog of `Polynomial.induction_on`. -/
+/--
+Similar to `MvPolynomial.induction_on` but only a weak form of `h_add` is required.
+In particular, this version only requires us to show
+that `motive` is closed under addition of nontrivial monomials not present in the support.
+-/
+@[elab_as_elim]
+theorem monomial_add_induction_on {motive : MvPolynomial σ R → Prop} (p : MvPolynomial σ R)
+    (C : ∀ a, motive (C a))
+    (monomial_add_weak :
```

Comment:
```suggestion
    (monomial_add :
```
I think I prefer this as branch name in the induction proofs... what do you think?